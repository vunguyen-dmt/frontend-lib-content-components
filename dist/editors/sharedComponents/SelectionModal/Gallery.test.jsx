import React from 'react';
import { shallow } from 'enzyme';

import { formatMessage } from '../../../testUtils';
import { Gallery } from './Gallery';

jest.mock('../../data/redux', () => ({
  selectors: {
    requests: {
      isFinished: (state, { requestKey }) => ({ isFinished: { state, requestKey } }),
    },
  },
}));

jest.mock('./GalleryCard', () => 'GalleryCard');

describe('TextEditor Image Gallery component', () => {
  describe('component', () => {
    const props = {
      galleryIsEmpty: false,
      searchIsEmpty: false,
      displayList: [{ id: 1 }, { id: 2 }, { id: 3 }],
      highlighted: 'props.highlighted',
      onHighlightChange: jest.fn().mockName('props.onHighlightChange'),
      intl: { formatMessage },
      isLoaded: true,
    };
    test('snapshot: not loaded, show spinner', () => {
      expect(shallow(<Gallery {...props} isLoaded={false} />)).toMatchSnapshot();
    });
    test('snapshot: loaded but no images, show empty gallery', () => {
      expect(shallow(<Gallery {...props} galleryIsEmpty />)).toMatchSnapshot();
    });
    test('snapshot: loaded but search returns no images, show 0 search result gallery', () => {
      expect(shallow(<Gallery {...props} searchIsEmpty />)).toMatchSnapshot();
    });
    test('snapshot: loaded, show gallery', () => {
      expect(shallow(<Gallery {...props} />)).toMatchSnapshot();
    });
    test('snapshot: not shot gallery', () => {
      const wrapper = shallow(<Gallery {...props} show={false} />);
      expect(wrapper.type()).toBeNull();
    });
  });
});
