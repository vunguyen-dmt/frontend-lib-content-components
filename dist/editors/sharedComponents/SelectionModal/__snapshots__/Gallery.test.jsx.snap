// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor Image Gallery component component snapshot: loaded but no images, show empty gallery 1`] = `
<div
  className="gallery p-4 bg-light-400"
  style={
    Object {
      "height": "375px",
      "margin": "0 -1.5rem",
    }
  }
>
  <FormattedMessage />
</div>
`;

exports[`TextEditor Image Gallery component component snapshot: loaded but search returns no images, show 0 search result gallery 1`] = `
<div
  className="gallery p-4 bg-light-400"
  style={
    Object {
      "height": "375px",
      "margin": "0 -1.5rem",
    }
  }
>
  <FormattedMessage
    defaultMessage="No search results."
    description="Label for when search returns nothing."
    id="authoring.selectionmodal.emptySearchLabel"
  />
</div>
`;

exports[`TextEditor Image Gallery component component snapshot: loaded, show gallery 1`] = `
<Scrollable
  className="gallery bg-light-400"
  style={
    Object {
      "height": "375px",
      "margin": "0 -1.5rem",
    }
  }
>
  <div
    className="p-4"
  >
    <SelectableBox.Set
      columns={1}
      name="images"
      onChange={[MockFunction props.onHighlightChange]}
      type="radio"
      value="props.highlighted"
    >
      <GalleryCard
        asset={
          Object {
            "id": 1,
          }
        }
        key="1"
        showId={false}
      />
      <GalleryCard
        asset={
          Object {
            "id": 2,
          }
        }
        key="2"
        showId={false}
      />
      <GalleryCard
        asset={
          Object {
            "id": 3,
          }
        }
        key="3"
        showId={false}
      />
    </SelectableBox.Set>
  </div>
</Scrollable>
`;

exports[`TextEditor Image Gallery component component snapshot: not loaded, show spinner 1`] = `
<div
  style={
    Object {
      "left": "50%",
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  <Spinner
    animation="border"
    className="mie-3"
    screenReaderText="loading..."
  />
</div>
`;
